extensions/amp-app-banner/amp-app-banner.md

```diff
@@ -0,0 +1,158 @@
+          <a href="https://amp-mk-3.herokuapp.com/examples/viewer.html">viewer.html (inside of viewer context — tap "Two" at the top of the page to show the page containing the banner)</a>
+        </li>
+      </ul>
+    </td>
+  </tr>
+</table>
+
+## Behavior
+`amp-app-banner` is a wrapper and minimal UI for a cross-platform, fixed-position banner showing a call-to-action to install an app. Includes conditional logic to direct to the right app on the right platform, and to hide permenantly if the user dismisses the banner.
+
+
+##Data Sources
+To extend and promote the usage of the natively supported app banners, we are using the exact data-sources the native app banners use on the respective platforms. iOS uses a `<meta name="apple-itunes-app">` tag in the head of the document and Android uses a `<link rel="manifest">`. 
+
+The AMP runtime parses the meta tag content attribute on iOS extracting the App ID and `app-argument` (usually used for deep-linking URIs - app-protocols like `whatsapp://` or `medium://`). On Android, the AMP runtimes makes an XHR request to fetch the `manifest.json` file, and parses its content to extract `app_id` from `related_applications` and it calculates the app store URL as well as open-in-app URL:
+
+```
+android-app://${appId}/${protocol}/${pathname}
+```
+
+##Appearance Behavior
+amp-app-banner provides no default UI and leave the UI to the developer. The developer can build any kind of UI inside the banner and style it accordingly. The only UI element provided by AMP and can’t be overridden by publishers is the dismiss button that allows users to dismiss the banner.
+
+One required UI element is the `button[open-link]` button, which is the click target for the banner to install the app, or open the deep-link if the app is already installed.
+
+###OS/Browser/Viewer Dependencies
+
+Because native app banners currently are not shown in the viewer context, `<amp-app-banner>` is shown inside of the viewer when the appropriate markup is present. Because system-level app banners are already shown in Android/Chrome and iOS/Safari OS/Browser combinations, `<amp-app-banner>` is hidden in those contexts to avoid redundancy.
+
+<table>
+  <tr>
+    <td></tc>
+    <td>Android + Chrome</tc>
+    <td>iOS + Safari</tc>
+    <td>Other OS + browser</tc>
+  </tr>
+  <tr>
+    <td>In AMP viewer</tc>
+    <td>Show amp-app-banner</tc>
+    <td>Show amp-app-banner</tc>
+    <td>Show amp-app-banner</tc>
+  </tr>
+  <tr>
+    <td>Outside of AMP viewer</tc>
+    <td>Show system banner</tc>
+    <td>Show system banner</tc>
+    <td>Show amp-app-banner</tc>
+  </tr>
+</table>
+
+
+##Dismissal Persistence
+The banner currently will be displayed always unless it was dismissed. Once dismissed the banner will never be displayed on that domain unless user visits on a different browser or clears their local storage.
+
+The persistence is managed through AMP Local Storage by storing a boolean value with a key `amp-app-banner:<element.id>`.
```

Comment:
This is not exactly true, and I don't think there's a need to expose this, this is implementation details, besides its not really stored like that in local storage (AMP Local Storage will encrypt and store an item per origin and inside that item will store different data including the amp-app-banner:<element.id>.

I'd say drop the `by storing a boolean value with a key....` part.